### PR TITLE
Add a GitHub Workflow for the Trivy security scanner

### PR DESCRIPTION
**Pull request by jof**
This also adds in a `.trivyignore` file for all of the existing, latent findings in the repository (regardless of actual impact).

Eventually, this new workflow will replace the existing `tfsec` job currently pulled into the "Lint (Terraform)" workflow.

Subsequently, we will go through, one-by-one and document exceptions or ignore in more targeted files/resources where possible with inline comments.

### CODE REVIEW

File: .github/workflows/trivy.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Trivy
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
+  pull_request:
+  merge_group:
+  workflow_dispatch:
```
Comment (wadells) [inline .github/workflows/trivy.yaml:10]: Do we need workflow dispatch here?  I understand the other 3, but I'm not sure what the motivation is for this trigger.

**Review by wadells — APPROVED** [review]

File: .github/workflows/trivy.yaml (hunk 1 above)
Comment (jof) [reply .github/workflows/trivy.yaml:10]: Cargo culting; I suppose it doesn't make much sense here.

**Review by wadells — APPROVED** [review]
Still LGTM.

**Review by adaadb6 — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Review by ravicious — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

File: .github/workflows/trivy.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Trivy
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
```
Comment (r0mant) [inline .github/workflows/trivy.yaml:7]: Shouldn't this be `branch/**`?

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-filters

File: .github/workflows/trivy.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  pull_request:
+  merge_group:
+
+jobs:
+  trivy:
+    uses: gravitational/shared-workflows/.github/workflows/trivy.yaml@main
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
```
Comment (r0mant) [inline .github/workflows/trivy.yaml:17]: Nit: Add trailing newline.

File: .github/workflows/trivy.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: Trivy
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
+  pull_request:
+  merge_group:
```
Comment (r0mant) [inline .github/workflows/trivy.yaml:9]: Is this supposed to run on each pull request and in the merge group? How long does this check take?

Comment (r0mant) [reply .github/workflows/trivy.yaml:9]: Looking at the history about 2 minutes, should be ok then.

**Review by r0mant — APPROVED** [review]

File: .github/workflows/trivy.yaml (hunk 2 above)
Comment (jof) [reply .github/workflows/trivy.yaml:7]: I cargo-culted this from other workflows in this repo. I presume it's meant to match the `branch/vNN` branches, but not anything with a slash in it (e.g. `branch/v10/something_else`); I don't see any of these, but it also seems like something we don't want to match.

File: .github/workflows/trivy.yaml (hunk 4 above)
Comment (jof) [reply .github/workflows/trivy.yaml:9]: Yes -- the goal is to eventually (once we've gained some comfort with it) make this check blocking merges.
The goal is to catch newly committed misconfigurations before they're merged.

File: .github/workflows/trivy.yaml (hunk 3 above)
Comment (jof) [reply .github/workflows/trivy.yaml:17]: Will fix.

**Pull request merged**